pkg/runtime/adoption_reconciler.go

```diff
@@ -224,10 +224,16 @@ func (r *adoptionReconciler) Sync(
 	}, described.RuntimeObject()); err != nil {
 		if apierrors.IsNotFound(err) {
 			// If Adopted AWS resource was not found in k8s, create it.
+
+			// Before creation, Keep the copy of original described object
+			// because after the create call, Status gets set to empty
+			describedCopy := described.DeepCopy()
 			if err := r.kc.Create(ctx, described.RuntimeObject()); err != nil {
 				return r.onError(ctx, desired, err)
 			}
-
+			// reset the status of described object to original value before
+			// making the Status Update call
+			described.SetStatus(describedCopy)
 			if err := r.kc.Status().Update(ctx, described.RuntimeObject()); err != nil {
```

Comment:
Why are we even calling `r.kc.Status().Update()` here? We just created the CR in the call above to `r.kc.Create()` so why are we immediately updating it here?

Reply:
afaik, spec and status are not applied at same time. you need separate call

Reply:
Adoption of resources that use the ARN as their primary identifier mean we have to place the status onto the object before it can be reconciled. We first call `.Create()`, with which we can only add the spec to the object. Then we need to call `.Status().Update()` to set the ARN.

Reply:
We need to update the status of new resource as well because for some resources the identifier is stored inside `Status.ACKResourceMetadata.ARN` . 
The Create call creates new Custom Resource with Spec and when `reconciler.go` tries to perform `ReadOne`, it fails with `NotFoundError`. NotFoundError is expected because ReadOne call expects a `Status.ACKResourceMetadata.ARN`. 

In non-adoption cases, this will result in resource creation but reconciler.go has a check that if NotFoundError occurs for AdoptedResource, Fail !!
That's why Sagemaker AdoptedModelPackage was never able reconcile from `reconciler.go` even when successfully reconciled inside `adoption_reconciler.go`

Reply:
thanks all. I always seem to forget the fact that there is no REPLACE operation on both spec and status at once...